runtime/jcl/common/stdinit.c

```diff
@@ -155,62 +145,72 @@ standardInit( J9JavaVM *vm, char *dllName)
 	}
 #endif /* !J9VM_INTERP_MINIMAL_JCL */
 #endif /* J9VM_OPT_SIDECAR */
 
-	vmFuncs->internalAcquireVMAccess(vmThread);
-
-	if (result == 0) {
-		U_32 runtimeFlags = computeJCLRuntimeFlags(vm);
-		result = initializeKnownClasses(vm, runtimeFlags);
-	}
-
-	if (result == 0) {
-		IDATA continueInitialization = TRUE;
-		/* Must do this before initializeAttachedThread */
-		vmFuncs->internalReleaseVMAccess(vmThread);
-
-		TRIGGER_J9HOOK_VM_INITIALIZE_REQUIRED_CLASSES_DONE(vm->hookInterface, vmThread, continueInitialization);
-		if (!continueInitialization) {
-			goto _fail;
+	if (JNI_OK == result) {
+		vmFuncs->internalAcquireVMAccess(vmThread);
+		
+		result = (jint)initializeRequiredClasses(vmThread, dllName);
+		
+		if (JNI_OK == result) {
+			result = vmFuncs->initializeHeapOOMMessage(vmThread);
 		}
+		
+		if (JNI_OK == result) {
+			U_32 runtimeFlags = computeJCLRuntimeFlags(vm);
+			result = initializeKnownClasses(vm, runtimeFlags);
+		}
+		
+		if (JNI_OK == result) {
+			IDATA continueInitialization = TRUE;
+			/* Must do this before initializeAttachedThread */
+			vmFuncs->internalReleaseVMAccess(vmThread);
+
+			TRIGGER_J9HOOK_VM_INITIALIZE_REQUIRED_CLASSES_DONE(vm->hookInterface, vmThread, continueInitialization);
+			if (!continueInitialization) {
+				goto _fail;
+			}
 
-		result = (jint)initializeSystemThreadGroup(vm, (JNIEnv *)vmThread);
-		if (result != JNI_OK) goto _fail;
+			result = (jint)initializeSystemThreadGroup(vm, (JNIEnv *)vmThread);
+			if (JNI_OK != result) {
+				goto _fail;
+			}
 
-#if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
-		vmFuncs->internalEnterVMFromJNI(vmThread);
-		vmFuncs->internalReleaseVMAccess(vmThread);
-#endif /* J9VM_INTERP_ATOMIC_FREE_JNI */
-		vmFuncs->initializeAttachedThread(vmThread, threadName, (j9object_t *)threadGroup, FALSE, vmThread);
+	#if defined(J9VM_INTERP_ATOMIC_FREE_JNI)
+			vmFuncs->internalEnterVMFromJNI(vmThread);
+			vmFuncs->internalReleaseVMAccess(vmThread);
+	#endif /* J9VM_INTERP_ATOMIC_FREE_JNI */
+			vmFuncs->initializeAttachedThread(vmThread, threadName, (j9object_t *)threadGroup, FALSE, vmThread);
 
-		vmFuncs->internalAcquireVMAccess(vmThread);
+			vmFuncs->internalAcquireVMAccess(vmThread);
 
-		if ((vmThread->currentException != NULL) || (vmThread->threadObject == NULL)) {
-			result = JNI_ERR;
-		} else {
-			vmFuncs->internalFindKnownClass(vmThread,
-				J9VMCONSTANTPOOL_JAVALANGTHREADDEATH,
-				J9_FINDKNOWNCLASS_FLAG_INITIALIZE | J9_FINDKNOWNCLASS_FLAG_NON_FATAL);
-			if (vmThread->currentException) {
+			if ((NULL != vmThread->currentException) || (NULL == vmThread->threadObject)) {
 				result = JNI_ERR;
+			} else {
+				vmFuncs->internalFindKnownClass(vmThread,
+					J9VMCONSTANTPOOL_JAVALANGTHREADDEATH,
+					J9_FINDKNOWNCLASS_FLAG_INITIALIZE | J9_FINDKNOWNCLASS_FLAG_NON_FATAL);
+				if (vmThread->currentException) {
+					result = JNI_ERR;
+				}
 			}
 		}
+		vmFuncs->internalReleaseVMAccess(vmThread);
+	} else {
+		goto _fail;
```

Comment:
This is not equivalent to the original code. Any one of the calls above could fail, and execution should not continue past here.

Reply:
I previously overlooked the failing cases inside the block. Already changed it back to the original one.